pandas/_libs/tslibs/timestamps.pyx

```diff
@@ -1469,6 +1470,22 @@ default 'raise'
             np.array([self.value], dtype='i8'), tz=self.tz)[0]
         return Timestamp(normalized_value).tz_localize(self.tz)
 
+    def strftime(self, format: str) -> str:
+        newformat = format
+        # only do additional processing if necessary
+        if self.nanosecond and '%f' in format:
+            newformat = []
+            for ch in format:
```

Comment:
Can you not just use a regex here instead?

Reply:
Yeah, OK that would probably be cleaner